### PR TITLE
Add support for special ordered sets

### PR DESCRIPTION
**Pull request by andreaslundell**

### CODE REVIEW

**Comment by andreaslundell** [comment]
As far as I can see, this works now for GAMS, AMPL and OSiL file types. I have tested with the files in MINLPLib with SOS-variables and the files you sent me @svigerske.

**Review by svigerske — APPROVED** [review]
Great!

File: src/MIPSolver/MIPSolverCbc.cpp [hunk 1]
```diff
@@ -380,6 +380,43 @@ int MIPSolverCbc::addLinearConstraint(
...
+        assert(variableWeights.size() == variableIndexes.size());
+
+        auto objects = new OsiObject*[1];
+        objects[0] = new OsiSOS(osiInterface.get(), variableIndexes.size(), &variableIndexes[0], &variableWeights[0],
+            (type == E_SOSType::One) ? 1 : 2);
+
+        osiInterface->addObjects(1, objects);
+
+        delete objects[0];
+        delete[] objects;
```
[the remark this answers is not in the document]
Comment (andreaslundell) [reply src/MIPSolver/MIPSolverCbc.cpp:404]: Sorry this did not work:

`error: invalid conversion from ‘OsiSOS**’ to ‘OsiObject**’` and I cannot (easily) make a cast... 

Was my commit actually a problem, or was your suggestion simply an improvement?

[the remark this answers is not in the document]
Comment (svigerske) [reply src/MIPSolver/MIPSolverCbc.cpp:404]: No, that was just an improvement.
Probably being explicit about the type,
```
OsiObject* object = new OsiSOS(...);
```
would work then.

[the remark this answers is not in the document]
Comment (andreaslundell) [reply src/MIPSolver/MIPSolverCbc.cpp]: That was the trick! Thx.

**Pull request merged**